packages/vx-shape/src/shapes/link/curve/LinkRadialCurve.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import { pointRadial } from 'd3-shape';
+import additionalProps from '../../../util/additionalProps';
+
+LinkRadialCurve.propTypes = {
+  innerRef: PropTypes.func
+};
+
+export default function LinkRadialCurve({
+  className,
+  innerRef,
+  data,
+  x = d => d.x,
+  y = d => d.y,
+  ...restProps
+}) {
+
+  const curve = (source, target) => {
+    const sa = x(source) - Math.PI / 2;
+    const sr = y(source);
+    const ta = x(target) - Math.PI / 2;
+    const tr = y(target);
+
+    const sc = Math.cos(sa);
+    const ss = Math.sin(sa);
+    const tc = Math.cos(ta);
+    const ts = Math.sin(ta);
+
+    const sx = sr * sc;
+    const sy = sr * ss;
+    const tx = tr * tc;
+    const ty = tr * ts;
+
+    const dx = tx - sx;
+    const dy = ty - sy;
+    const ix = 0.2 * (dx + dy);
+    const iy = 0.2 * (dy - dx);
```

Comment:
should the `0.2` be configurable here?

Reply:
I played around with this but I didn't see much benefit as this is being built using [cubic bezier curves](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Bezier_Curves) (`C` path command) that I ported from Vega.  We would probably need to expose both of these `0.2` separately, but I wasn't sure what to call them, etc.

I took a quick stab at using a simpler quadratic curve (`Q` path command) where a single "midpoint" would define the curve but I was having trouble creating the same curve.

I've kicked around the idea in my head a bit and part of me thinks exposing `<HorizontalLink curve={curve} />`, `<VerticalLink curve={curve} />` and `<RadialLink curve={curve} />` and allow passing the curve function similar to show [d3-shape](https://github.com/d3/d3-shape#curves) does this.  This is basically what my `line` functions are doing, such as...

```js
const step = (source, target) => {
    const sa = x(source) - Math.PI / 2;
    const sr = y(source);
    const ta = x(target) - Math.PI / 2;
    const tr = y(target);

    const sc = Math.cos(sa);
    const ss = Math.sin(sa);
    const tc = Math.cos(ta);
    const ts = Math.sin(ta);
    const sf = Math.abs(ta - sa) > Math.PI ? ta <= sa : ta > sa;
    
    return `
      M${sr * sc},${sr * ss}
      A${sr},${sr} 0 0,${sf ? 1 : 0}
      ${sr * tc},${sr * ts}
      L${tr * tc},${tr * ts}
    `;
  };
```

I haven't looked to much into the details on how to implement this (looks like you you're supposed to use the [custom curves](https://github.com/d3/d3-shape#custom-curves) api).  This might also be overkill for 2 points...

I still think exposing the simple `LinkHorizontalStep` is the way to go, this is more for the customization beyond the defaults...